### PR TITLE
Plugin search hints: various fixes and enhancements

### PR DESCRIPTION
**Pull request by eliorivero**
- [x] fix link to configure settings when module is already active
- [x] remove condition checking for dev mode since this only works when Jetpack is connected
- [x] removed condition checking whether Jetpack is connected since it's the first thing checked before initializing the class.
- [x] fix site fragment missing in outbound "Purchase" links
- [x] incorporate Malware scanning tags into VP module header
- [x] rename VP module name from “Data Backups” to “Backups and Scanning”
- [x] add terms to VP `backups, off-site backups, offsite backup, offsite, off-site`
- [x] remove Akismet, VaultPress and Jetpack plugin cards when a hint is displayed
- [x] always display a "Get started" button for Akismet and VaultPress cards
- [x] redo matching algorithm so it does partial matching. For example, matching 'delivery' in 'content delivery network', or 'lazy image' in 'lazy images'
- [x] reduce the features array to the whitelisted features as soon as possible for better performance

### CODE REVIEW

**Review by jeherve — DISMISSED** [review]
I gave it a try and would have a few remarks:

------

When on a free plan, I am still offered the option to "enable" "Jetpack: Backups and Scanning". I think it should say "Purchase" instead, and link either to the plans page or to the feature page on Jetpack.com.

image

clicking on "enable" enables the VaultPress module, but since that's not really a regular module it does not do much.

------

On a site where the Akismet plugin is installed, but not activated, I get the "Get Started" option in the card. Would we be able to change that to activate or enable, and activate the plugin behind the scenes?

image

------

It seems you've included a `package-lock.json` file in there; it can probably be removed.

File: modules/plugin-search.php
Comment (eliorivero) [reply modules/plugin-search.php]: I was thinking we might have to do the same with VaultPress, that's why I didn't hardcode it. I'm going to discuss this but maybe we'll end up doing it.

File: modules/vaultpress.php
Comment (eliorivero) [reply modules/vaultpress.php]: Yup. I relayed this feedback and will update when we have a new text.

File: modules/plugin-search.php
Comment (kraftbj) [inline modules/plugin-search.php]: Not a huge fan of escaping within this function instead of right here on display as it could introduce an issue if `get_configure_url` is refactored in the future and we don't patch up the escaping right, but since it is escaped I don't want to block on that.

**Review by kraftbj — DISMISSED** [review]
One nitpicky thing, but not critical enough to block. I'd like a second opinion on that, but marking approving if the second opinion isn't from Crew.

**Review by kraftbj — APPROVED** [review]

**Comment by jeherve** [comment]
Cherry-picked to `branch-7.1` in 9e583bc196c9ab319b5f2f82ad3074dfa916817b

**Pull request merged**
